src/libratbag-hidraw.c

```diff
@@ -1675,3 +1675,55 @@ ratbag_hidraw_read_input_report_index(struct ratbag_device *device, uint8_t *buf
 
 	return rc >= 0 ? rc : -errno;
 }
+
+int
+ratbag_hidraw_read_input_report_filtered(struct ratbag_device *device, uint8_t *buf, size_t len,
+				 ratbagd_hidraw_filter_t filter)
+{
+	return ratbag_hidraw_read_input_report_index_filtered(device, buf, len, 0, filter);
+}
+
+int
+ratbag_hidraw_read_input_report_index_filtered(struct ratbag_device *device, uint8_t *buf, size_t len,
+				 int hidrawno, ratbagd_hidraw_filter_t filter)
```

Comment:
`ratbag_hidraw_read_input_report_index` should be able to call this function with a `NULL` filter so we can re-use that code.

Or, even better since all of libratbag is effectively internal API, you could just change all callers of `ratbag_hidraw_read_input_report_index()` to pass NULL as filter so we only have one function.

Reply:
hm, yeah reusing this makes sense, simplifies things